CMakeLists.txt

```diff
@@ -13,9 +13,28 @@ endif()
 find_package(ament_cmake REQUIRED)
 find_package(Eigen3 REQUIRED)
 
-install(DIRECTORY include/${PROJECT_NAME}/
-        DESTINATION include/${PROJECT_NAME}
-        FILES_MATCHING PATTERN "*.h")
+add_library(${PROJECT_NAME} INTERFACE
+  include/${PROJECT_NAME}/eigen_stl_containers.h
+  include/${PROJECT_NAME}/eigen_stl_map_container.h
+  include/${PROJECT_NAME}/eigen_stl_vector_container.h
+)
+
+ament_target_dependencies(${PROJECT_NAME} Eigen3)
+ament_export_targets(${PROJECT_NAME}Targets HAS_LIBRARY_TARGET)
```

Comment:
IIRC `HAS_LIBRARY_TARGET` means "has a shared library target", and causes an environment hook for `LD_LIBRARY_PATH` to be set. It shouldn't be set here because only an interface target is being used.

Reply:
LD_LIBRARY_PATH has nothing to do with it. The "library" is declared as [INTERFACE](https://cmake.org/cmake/help/latest/command/add_library.html#id4) and therefore is header only

Reply:
> LD_LIBRARY_PATH has nothing to do with it.

`HAS_LIBRARY_TARGET` causes `LD_LIBRARY_PATH` to be set when the workspace containing this package is sourced. We don't need `LD_LIBRARY_PATH` because, as you say, this library is header only.

https://github.com/ament/ament_cmake/blob/3f305a3e1e51f2bd4bd4bf5525a1981c34ff5d08/ament_cmake_export_targets/cmake/ament_export_targets.cmake#L22-L24

https://github.com/ament/ament_cmake/blob/3f305a3e1e51f2bd4bd4bf5525a1981c34ff5d08/ament_cmake_export_targets/cmake/ament_export_targets.cmake#L43-L48

https://github.com/ament/ament_cmake/blob/3f305a3e1e51f2bd4bd4bf5525a1981c34ff5d08/ament_cmake_export_libraries/ament_cmake_export_libraries-extras.cmake#L18-L29